cmd/schema_gen.go

```diff
@@ -0,0 +1,195 @@
+			err = WriteYMLToFile(ymlPath, ymlBody)
+			if err != nil {
+				fmt.Println("Error writing YML to file in path")
+				os.Exit(1)
+			}
+			fmt.Println("\n✅ " + modelName + "schema successfully written to path: " + ymlPath)
+
+		} else {
+			fmt.Println("\n🔍 " + modelName + " schema file found.. 🛠  Updating schema file")
+
+			// set working schema model to current schema model
+			schemaModel := model.Schema
+			// add and remove columns in-place
+			addMissingColumnsToSchema(schemaModel, bqColumns)
+			removeOutdatedColumnsFromSchema(schemaModel, bqColumns)
+			schemaFile.Models = properties.Models{schemaModel}
+
+			ymlBody := marshalYML(schemaFile)
+			err = WriteYMLToFile(ymlPath, ymlBody)
+			if err != nil {
+				fmt.Println("Error writing YML to file in path")
+				os.Exit(1)
+			}
+			fmt.Println("\n✅ " + modelName + "schema successfully updated at path: " + ymlPath)
+		}
+	},
+}
+
+func marshalYML(schema interface{}) string {
+	y, err := yaml.Marshal(&schema)
+	if err != nil {
+		fmt.Println("Could not marshal schema")
+	}
+	yml := fmt.Sprintf("\n%s\n\n", string(y))
+	return yml
+}
+
+func GetColumnsForModel(modelName string, target *config.Target) ([]string, error) {
+	schema, err := bigquery.GetColumnsFromTable(modelName, target)
+	if err != nil {
+		return nil, err
+	}
+
+	columns := []string{}
+	for _, FieldSchema := range schema {
+		column := fmt.Sprintf("%v", FieldSchema.Name)
+		columns = append(columns, column)
+	}
+	return columns, nil
+}
+
+func WriteYMLToFile(filePath string, body string) error {
+	f, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	_, _ = f.WriteString("# Auto-generated by DDBT at " + time.Now().String())
+	_, err = f.WriteString(body)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// generate an empty schema file which will be populated according to existing yml schemas and the bigquery schema.
+// Returns the local path for the yml file and the yml file struct
+func generateEmptySchemaFile(model *fs.File) (ymlPath string, schemaFile properties.File) {
+	ymlPath = strings.Replace(model.Path, ".sql", ".yml", 1)
+	schemaFile = properties.File{}
+	schemaFile.Version = properties.FileVersion
+	return ymlPath, schemaFile
+}
+
+// generate a new schema model for the provided model name and bqcolumns
+// this is used when then is no existing model
+func generateNewSchemaModel(modelName string, bqColumns []string) properties.Model {
+	schemaModel := properties.Model{}
+	schemaModel.Name = modelName
+	schemaModel.Description = "Please fill this in with a useful description.."
+	schemaModel.Columns = []properties.Column{}
+	for _, bqCol := range bqColumns {
+		column := properties.Column{}
+		column.Name = bqCol
+		schemaModel.Columns = append(schemaModel.Columns, column)
+	}
+
+	return schemaModel
+}
+
+// check if bq column is in schema (add missing)
+func addMissingColumnsToSchema(schemaModel *properties.Model, bqColumns []string) {
```

Comment:
For these pure methods here (i.e. functions which only operate on their given input and will always do the same thing) could we add some unit tests?

Reply:
@DomBlack should I add these as method on the Model type in propertiesFile.go and then add unit tests toe propertiesFile_test.go? 

I noticed there aren't any tests in the cmd package

Reply:
@robertpknight I'd add a new `schema_gen_test.go` and test there there.

For now I don't think we need to move them to the propertiesFile struct as they're very specific to this usecase .